docs/changelog/115932.yaml

```
@@ -0,0 +1,5 @@
+pr: 115932
+summary: "[8.x] Limit the number of tasks that a single search can submit"
```

Nit-picking: the number of tasks is still the same, we just submit them differently to keep the total contribution to the queue size under control?

```suggestion
summary: "[8.x] Limit the contribution to the search queue size of a single shard"
```

---

I guess it depends on how you look at it. It is true that Lucene will still create the tasks, but Elasticsearch now conditionally passes them through to the executor. As a result, the number of tasks observed in the search thread pool stats is much lower, hence I think it is accurate to word it as "limit the number of tasks". because in fact we don't submit such tasks at all, we just execute them in the caller thread instead to prevent rejections.

---

Fair, this should not block merging.